src/Exercise/TicketPlease.elm

```diff
@@ -0,0 +1,115 @@
+    , assignTicketUsesRecordUpdate
+    , assignedToDevTeamArgumentDestructureInCase
+    , assignedToDevTeamArgumentShouldUseNamedAndRecord
+    , emptyCommentArgumentShouldUseTupleAndIgnore
+    , numberOfCreatorCommentsArgumentShouldUseNamedAndRecord
+    , numberOfCreatorCommentsDestructuresTupleInLetAndLambda
+    , ruleConfig
+    )
+
+import Analyzer exposing (CalledExpression(..), CalledFrom(..), Find(..), Pattern(..))
+import Comment exposing (Comment, CommentType(..))
+import Dict
+import Elm.Syntax.Pattern exposing (Pattern(..))
+import Review.Rule exposing (Rule)
+import RuleConfig exposing (AnalyzerRule(..), RuleConfig)
+
+
+ruleConfig : RuleConfig
+ruleConfig =
+    { slug = Just "ticket-please"
+    , restrictToFiles = Just [ "src/TicketPlease.elm" ]
+    , rules =
+        [ CustomRule emptyCommentArgumentShouldUseTupleAndIgnore
+            (Comment "emptyComment argument doesn't destructure a tuple and wild card" "elm.ticket-please.destructure_emptyComment_argument" Essential Dict.empty)
+        , CustomRule numberOfCreatorCommentsArgumentShouldUseNamedAndRecord
+            (Comment "numberOfCreatorComments argument doesn't destructure a record in a named pattern" "elm.ticket-please.destructure_numberOfCreatorComments_argument" Essential Dict.empty)
+        , CustomRule numberOfCreatorCommentsDestructuresTupleInLetAndLambda
+            (Comment "numberOfCreatorComments doesn't destructure in let and lambda" "elm.ticket-please.destructure_numberOfCreatorComments_expressions" Essential Dict.empty)
+        , CustomRule assignedToDevTeamArgumentShouldUseNamedAndRecord
+            (Comment "assignedToDevTeam argument doesn't destructure a record in a named pattern" "elm.ticket-please.destructure_assignedToDevTeam_argument" Essential Dict.empty)
+        , CustomRule assignedToDevTeamArgumentDestructureInCase
+            (Comment "assignedToDevTeam argument doesn't destructure in a case block" "elm.ticket-please.destructure_assignedToDevTeam_expressions" Essential Dict.empty)
+        , CustomRule assignTicketToArgumentShouldUseNamedRecordAndAs
+            (Comment "assignTicketTo argument doesn't destructure a record in a named pattern using as" "elm.ticket-please.destructure_assignTicketTo_argument" Essential Dict.empty)
+        , CustomRule assignTicketToArgumentShouldUseIgnore
+            (Comment "assignTicketTo argument doesn't ignore cases" "elm.ticket-please.assignTicketTo_ignore_cases" Actionable Dict.empty)
+        , CustomRule assignTicketUsesRecordUpdate
+            (Comment "assignTicketTo doesn't use the record update syntax" "elm.ticket-please.assignTicketTo_use_record_update" Actionable Dict.empty)
+        ]
+    }
+
+
+emptyCommentArgumentShouldUseTupleAndIgnore : Comment -> Rule
+emptyCommentArgumentShouldUseTupleAndIgnore =
+    Analyzer.functionCalls
+        { calledFrom = TopFunction "emptyComment"
+        , findExpressions = [ ArgumentWithPattern Tuple, ArgumentWithPattern Ignore ]
```

Comment:
I didn't fully understand the changes in the analyser, but the end result is good here, this is very simple and easy to write.

Reply:
Thank you, that was my goal :)

In a nutshell, the analyzer now checks for patterns in pattern matching, the tricky part is that patterns in the function declaration arguments and patterns in expressions (let, case, lambda) live in different parts of the AST so I had to complicate the traversal and matching logic.